lightning/src/chain/channelmonitor.rs

```diff
@@ -295,21 +312,29 @@ impl<ChanSigner: ChannelKeys, C: Deref, T: Deref, F: Deref, L: Deref> ChainMonit
 	/// Updates the monitor which monitors the channel referred to by the given outpoint.
 	pub fn update_monitor(&self, outpoint: OutPoint, update: ChannelMonitorUpdate) -> Result<(), MonitorUpdateError> {
 		let mut monitors = self.monitors.lock().unwrap();
-		match monitors.get_mut(&outpoint) {
-			Some(orig_monitor) => {
-				log_trace!(self.logger, "Updating Channel Monitor for channel {}", log_funding_info!(orig_monitor));
-				orig_monitor.update_monitor(update, &self.broadcaster, &self.logger)
-			},
-			None => Err(MonitorUpdateError("No such monitor registered"))
+		if let Some(orig_monitor) = monitors.get_mut(&outpoint) {
+			log_trace!(self.logger, "Updating Channel Monitor for channel {}", log_funding_info!(orig_monitor));
+			if let Err(e) = orig_monitor.update_monitor(&update, &self.broadcaster, &self.logger) {
+				return Err(e)
+			}
+			match self.data_persister.update_channel_data(outpoint, &update, orig_monitor) {
```

Comment:
Now that we're doing a potentially-multi-disk update here - we need to change the return type to ChannelMonitorUpdateErr so that we can communicate back the Permanent/Temporary distinction.